pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -26,13 +30,28 @@ func (ca *cloudCIDRAllocator) performMultiNetworkCIDRAllocation(node *v1.Node, i
 	if err != nil {
 		return nil, fmt.Errorf("node=%s error fetching networks: %v", node.Name, err)
 	}
+
+	// get networks from Node's network-status annotation
+	nodeUpNet, err := getUpNetworks(node)
+
+	// statusNetworks is map of networks that are Ready and status==UP
+	statusNetworks := make(map[string]struct{})
+	// networks is list of networks that are Ready
 	networks := make([]*networkv1.Network, 0)
-	// ignore networks that are under deletion.
+	// filter networks based on Ready condition
 	for _, network := range k8sNetworksList {
-		if network.DeletionTimestamp.IsZero() {
+		if meta.IsStatusConditionTrue(network.Status.Conditions, string(networkv1.NetworkConditionStatusReady)) || networkv1.IsDefaultNetwork(network.Name) {
 			networks = append(networks, network)
+			_, ok := nodeUpNet[network.Name]
+			if ok {
+				// we do not care about status of default Network, since it has to be
+				// always available
```

Comment:
The comment seems to be at the wrong place. If I read this code correctly, the default network will never get here since nodeUpNet will never have the default network. We should move this comment out of the if statement.

Reply:
I've modified the comment to be more clear. Let me know if that works.